src/Parser/MarksmanshipHunter/Modules/FocusChart/Focus.js

```diff
@@ -0,0 +1,374 @@
+    start: PropTypes.number.isRequired,
+    end: PropTypes.number.isRequired,
+  };
+  constructor() {
+    super();
+    this.state = {
+      focusData: null,
+      bossHealth: null,
+    };
+	
+  }
+  
+
+  componentWillMount() {
+    this.load(this.props.reportCode, this.props.actorId, this.props.start, this.props.end);
+  }
+  componentWillReceiveProps(newProps) {
+    if (newProps.reportCode !== this.props.reportCode || newProps.actorId !== this.props.actorId || newProps.start !== this.props.start || newProps.end !== this.props.end) {
+      this.load(newProps.reportCode, newProps.actorId, newProps.start, newProps.end);
+    }
+  }
+  load(reportCode, actorId, start, end) {
+    const focusPromise = fetch(makeWclUrl(`report/tables/resources/${reportCode}`, {
+      start,
+      end,
+      sourceid: actorId,
+      abilityid: 102,
+    })) 
+      .then(response => response.json())
+      .then((json) => {
+        if (json.status === 400 || json.status === 401) {
+          throw json.error;
+        } else {
+          this.setState({
+            focusData: json,
+          });
+        }
+      });
+	const playerPromise = fetch(makeWclUrl(`report/events/${reportCode}`, { //We really just grab for player stats- I'm sure there's a better way
+      start,
+      end,
+      actorid: actorId,
+    }))
+      .then(response => response.json())
+      .then((json) => {
+        if (json.status === 400 || json.status === 401) {
+          throw json.error;
+        } else {
+           this.setState({
+            playerData: json,
+          });
+        }
+      });	
+    const bossHealthPromise = fetch(makeWclUrl(`report/tables/resources/${reportCode}`, {
+      start,
+      end,
+      sourceclass: 'Boss',
+      hostility: 1,
+      abilityid: 1000,
+    }))
+      .then(response => response.json())
+      .then((json) => {
+        if (json.status === 400 || json.status === 401) {
+          throw json.error;
+        } else {
+          this.setState({
+            bossHealth: json,
+          });
+        }
+      });
+
+    return Promise.all([focusPromise, bossHealthPromise, playerPromise]);
+  }
+
+  render() {
+    if (!this.state.focusData || !this.state.bossHealth || !this.state.playerData) {
+      return (
+        <div>
+          Loading...
+        </div>
+      );
+    }
+	const actorId = this.props.actorId;
+	
+	let maxFocus = 0;
+	let found = false;
+	let ii = 0;
+	while (!found){ //determine max focus (max focus is variable)
+		if (this.state.playerData.events[ii].sourceID === actorId &&(this.state.playerData.events[ii].type === 'cast' || this.state.playerData.events[ii].type === 'energize') ){
+			maxFocus = this.state.playerData.events[ii].classResources[0]['max'];
+			found = true;
+		}
+		else{
+			ii++;
+		}
+	}
+	
+	const focusGen = Math.round((10 + .1 * this.state.playerData.events[0].hasteRanged / 375)*100)/100; //TODO: replace constant passive FocusGen (right now we don't account for lust/hero or Trueshot)
```

Comment:
If you pass the parser to this component on creation (in your CombatLogParser), you can get this with `parser.combatants.selected.hasteRating`.